### DISCUSSION TITLE
osu!lazer touchscreen implementation causes a problem in osu! mode

### DISCUSSION THREAD

**Discussion by timelessnesses** [ppy/osu#22729] [Q&A] [answered]
If you somehow press the key button press (the one that isn't intended to control the cursor I will call it "key input") before you control the cursor then it would only focus the "key input" and switch back and forth and the cursor pointer (their intended finger press that is supposed to control cursor) will now act as "key input".  
This can be very frustrating when you playing and you tap jumps and streams altogether.  
Although I find out that you can also drag your finger across the screen like a pen on a drawing tablet, I find that  actually creating friction on my finger causing it to be very hurt at the end of the map.  
osu!droid fix this by accepting every touch input as a cursor then send those touch input as a mouse input to every of those cursors.

**Comment by Axilotl17** [comment, marked as answer]
> If you somehow press the key button press (the one that isn't intended to control the cursor I will call it "key input") before you control the cursor then it would only focus the "key input" and switch back and forth and the cursor pointer (their intended finger press that is supposed to control cursor) will now act as "key input".

#22556

> I find that actually creating friction on my finger causing it to be very hurt at the end of the map.

there is no solution to this issue. there is no other way to detect input. use a stylus if you want. 

> osu!droid fix this by accepting every touch input as a cursor then send those touch input as a mouse input to every of those cursors.

there might be a toggle to use this mode in the future, I'm not entirely sure.

also, it's called osu! standard, or std; not "osu! mode"

**Reply by Axilotl17** [reply]
also please don't do a bunch of "issues" in one discussion, split them up please.

**Comment by lolihanter1228** [comment]
in the wasp laser, during editing and creating maps on phones, it is impossible to create either a slider or a spinner, a maximum note, i.e. a button, please fix this problem

**Reply by Axilotl17** [reply]
> also please don't do a bunch of "issues" in one discussion, split them up please.

??? please read what i wrote. It's literally marked as answer. Make a new issue/discussion.
